### PR TITLE
Sudoku puzzle is developed

### PR DESCRIPTION
**Pull request by ANKITy102**
issue #690 

**Description**
I have developed and thoroughly tested the Sudoku Puzzle, and it's fully functional and ready for review.

**Game Features:**

- A 9x9 Sudoku grid is displayed at the start of the game, divided into 3x3 subgrids for a classic Sudoku experience.
- The initial Sudoku grid is partially filled with numbers, while the remaining cells are left blank for the user to fill in.
- Clear and concise game instructions are provided on the sidebar to guide users through the gameplay.
- Users have the option to reveal the solution by clicking on a dedicated button, which serves as a helpful tool for improving their Sudoku-solving skills.
- Players can reset the Sudoku board at any time to start a new game with a fresh and randomized puzzle.

<h2> Screenshots </h2>

Screenshot 2023-10-08 024700
Screenshot 2023-10-08 024710
Screenshot 2023-10-08 030153

<h3>I am in the process of creating a pull request for this feature. Your feedback is welcome!</h3>

<hr/>

**Status**

- [x] Code changes have been tested against prettier, or there are no code changes
- [x] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**

- [ ] This PR changes the codebase
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
  - [ ] This PR changes the internal workings with no modifications to the external API (bug fixes, performance improvements)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by tobySolutions** [comment]
Thank you very much @ANKITy102. Reviewing now.

**Comment by Timonwa** [comment]
This looks awesome @ANKITy102 . nice work.

**Comment by ANKITy102** [comment]
@tobySolutions 
Could you please review and consider merging my pull request? Thank you!

**Comment by Timonwa** [comment]
Is It possible for you to distinguish each block with grid lines like so:
download

**Comment by tobySolutions** [comment]
Hey @ANKITy102, can you send a screen recording of you playing the game, I don't think the game works 🤔

**Comment by Drex72** [comment]
@ANKITy102 I think there's an issue with your logic for the sudoku solver
could you please take a second look at it and then make a pr

**Comment by ANKITy102** [comment]
> @ANKITy102 I think there's an issue with your logic for the sudoku solver
> could you please take a second look at it and then make a pr

@Drex72 
Okay! I will fix it by today. Thank you:)

**Comment by ANKITy102** [comment]
> Hey @ANKITy102, can you send a screen recording of you playing the game, I don't think the game works 🤔 

Sorry @tobySolutions 
I got stuck in other work but I will share the screen recording of the game by today. Sorry for the delay.

**Comment by Drex72** [comment]
> > @ANKITy102 I think there's an issue with your logic for the sudoku solver
> > could you please take a second look at it and then make a pr
> 
> @Drex72 Okay! I will fix it by today. Thank you:)

No Problem, would be expecting it
i would really love to see this merged to the main branch

**Comment by ANKITy102** [comment]
> Is It possible for you to distinguish each block with grid lines like so: download
<hr/>
@Timonwa
 
### I have added the borders to make each block distinguishable with grid lines.  Below is the image for the same.
<br/>

image


### Thank you for your feedback!

**Comment by ANKITy102** [comment]
> @ANKITy102 I think there's an issue with your logic for the sudoku solver could you please take a second look at it and then make a pr


@Drex72

I've updated the game logic, and it's now working correctly. I've thoroughly tested the Sudoku puzzle against the three essential conditions for validity:

1. Each row contains the digits 1-9 without repetition.
2. Each column contains the digits 1-9 without repetition.
3. Each of the nine 3x3 sub-boxes in the grid contains the digits 1-9 without repetition.

I'm pleased to report that the puzzle passes all three conditions. 
### Thank you for your valuable feedback!

**Comment by ANKITy102** [comment]
@tobySolutions 

I've created a screen recording demonstrating me playing the Sudoku puzzle game. I hope this helps clarify any issues you may have encountered.

https://github.com/aidenybai/million/assets/119873198/5ce72362-cedc-46e2-9d15-1e35b263963a






### Thank you for your feedback, and I look forward to your thoughts.

**Comment by tobySolutions** [comment]
Awesome @ANKITy102, checking it out now.

File: packages/kitchen-sink/src/examples/sudoku.tsx [hunk 1]
```diff
@@ -0,0 +1,504 @@
...
+
+ 
+  return (
+    <main
+      style={{
+        background:'linear-gradient(to bottom right, rgb(40, 40, 43),rgb(48, 25, 52))',
+        position : 'relative'
+      }}
+    >
+      <style>
```
Comment (tobySolutions) [inline packages/kitchen-sink/src/examples/sudoku.tsx:185]: Your styles are kind of causing a problem. If you could review this @ANKITy102; 

https://github.com/aidenybai/million/assets/96334363/cf75d13e-f7c0-46a6-8884-87dca0069bc1

**Comment by Drex72** [comment]
> > @ANKITy102 I think there's an issue with your logic for the sudoku solver could you please take a second look at it and then make a pr
> 
> @Drex72
> 
> I've updated the game logic, and it's now working correctly. I've thoroughly tested the Sudoku puzzle against the three essential conditions for validity:
> 
> 1. Each row contains the digits 1-9 without repetition.
> 2. Each column contains the digits 1-9 without repetition.
> 3. Each of the nine 3x3 sub-boxes in the grid contains the digits 1-9 without repetition.
> 
> I'm pleased to report that the puzzle passes all three conditions.
> 
> ### Thank you for your valuable feedback!

That's Awesome
Just resolve the issues that was commented on by @tobySolutions  and it should be good to go

File: packages/kitchen-sink/src/examples/sudoku.tsx (hunk 1 above)
Comment (ANKITy102) [reply packages/kitchen-sink/src/examples/sudoku.tsx:185]: @tobySolutions 
I have fixed the issue. You can verify. Thank you!

Comment (ANKITy102) [reply packages/kitchen-sink/src/examples/sudoku.tsx:185]: > Your styles are kind of causing a problem. If you could review this @ANKITy102;
> 
>  Screen.Recording.2023-10-14.at.20.41.14.mov

@tobySolutions 
I have fixed the styling issue. Below is the video for the same

https://github.com/aidenybai/million/assets/119873198/3726d538-43ef-4cad-8824-0f83a3f540cc

**Comment by ANKITy102** [comment]
@tobySolutions 
Could you please review my pr? 
Thank you!

**Comment by ANKITy102** [comment]
@tobySolutions 
I have fixed the styling issue. Below is the video for the same.

https://github.com/aidenybai/million/assets/119873198/677f6d0d-8082-4afc-8dad-a527859ae50e

**Comment by tobySolutions** [comment]
> @tobySolutions I have fixed the styling issue. Below is the video for the same.
> 
>  sudoku_style_fixed.mp4

Awesome! Thanks @ANKITy102. I'm reviewing it now.

**Review by tobySolutions — CHANGES_REQUESTED** [review]
Could you maybe review the mobile view of this? The mobile doesn't look too good.


Uploading Screen Recording 2023-10-19 at 22.22.03.mov…

**Pull request closed without merging**